proving-system/stark/src/constraints/boundary.rs

```diff
@@ -0,0 +1,117 @@
+use lambdaworks_math::{
+    field::{element::FieldElement, traits::IsField},
+    polynomial::Polynomial,
+};
+
+pub(crate) struct BoundaryConstraint<FE> {
+    col: usize,
+    step: usize,
+    value: FE,
+}
+
+impl<F: IsField> BoundaryConstraint<FieldElement<F>> {
+    #[allow(dead_code)]
+    pub(crate) fn new(col: usize, step: usize, value: FieldElement<F>) -> Self {
+        Self { col, step, value }
+    }
+
+    pub(crate) fn new_simple(step: usize, value: FieldElement<F>) -> Self {
+        Self {
+            col: 0,
+            step,
+            value,
+        }
+    }
+}
+
+pub(crate) struct BoundaryConstraints<FE> {
+    constraints: Vec<BoundaryConstraint<FE>>,
+}
+
+impl<F: IsField> BoundaryConstraints<FieldElement<F>> {
+    #[allow(dead_code)]
+    pub(crate) fn new() -> Self {
+        Self {
+            constraints: Vec::<BoundaryConstraint<FieldElement<F>>>::new(),
+        }
+    }
+
+    pub(crate) fn from_constraints(constraints: Vec<BoundaryConstraint<FieldElement<F>>>) -> Self {
+        Self { constraints }
+    }
+
+    pub(crate) fn get_steps(&self) -> Vec<usize> {
+        let mut steps = Vec::with_capacity(self.constraints.len());
+        for constraint in self.constraints.iter() {
+            steps.push(constraint.step);
+        }
+
+        steps
```

Comment:
This could also be done with
```suggestion
        self.constraints.iter().map(|constraint| constraint.step).collect()
```